src/components/books/categories.js

```diff
@@ -1,10 +1,28 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
 
 function Category() {
+  const { category } = useSelector((state) => state.category);
+  const [searchField, setSearchField] = useState('');
+  const [filterResult, setFilterResult] = useState(category);
+
+  const filterHandler = () => {
+    setFilterResult(
+      filterResult.map((item) => item.country.includes(searchField)),
+    );
+  };
+
   return (
-    <button className="cat" type="button">
-      Check status
-    </button>
+    <div className="categories">
+      <input
+        type="text"
+        value={searchField}
+        onChange={(e) => setSearchField(e.target.value)}
+      />
+      <button className="cat" type="button" onClick={filterHandler}>
+        Check status
+      </button>
```

Comment:
- [ ] Kindly make sure when the check status button is clicked, it dispatches the  `underConstruction` from your action creator in categorySlice.js